extensions/VisualEditor/wikia/modules/ve/dm/ve.dm.WikiaInfoboxSpecModel.js

```diff
@@ -0,0 +1,246 @@
+/*!
+ * VisualEditor DataModel WikiaInfoboxSpecModel class.
+ */
+
+/**
+ * MediaWiki template specification.
+ *
+ * @class
+ *
+ * @constructor
+ * @param template TemplateModel
+ */
+ve.dm.WikiaInfoboxSpecModel = function VeDmWikiaInfoboxSpecModel( template ) {
+	// Properties
+	this.template = template;
+	this.params = {};
+	this.paramOrder = [];
+	this.groups = [];
+
+	// Initialization
+	this.fill();
+};
+
+/* Methods */
+
+/**
+ * Extend with template spec data.
+ *
+ * Template spec data is available from the TemplateData extension's API. Extension is passive so
+ * any filled in values are not overwritten unless new values are available. This prevents changes
+ * in the API or fill methods from causing issues.
+ *
+ * @param {Object} data Template spec data
+ * @param {string} [data.description] Template description
+ * @param {string[]} [data.paramOrder] Canonically ordered parameter names
+ * @param {Object} [data.params] Template param specs keyed by param name
+ * @param {string[][]} [data.groups] Lists of param groups
+ */
+ve.dm.WikiaInfoboxSpecModel.prototype.extend = function ( data ) {
+	var key, param;
+
+	if ( Array.isArray( data.paramOrder ) ) {
+		this.paramOrder = data.paramOrder.slice();
+	}
+	if ( ve.isPlainObject( data.params ) ) {
+		for ( key in data.params ) {
+			// Pre-fill spec
+			if ( !this.params[key] ) {
+				this.params[key] = this.getDefaultParameterSpec( key );
+			}
+			param = this.params[key];
+			// Extend existing spec
+			ve.extendObject( true, this.params[key], data.params[key] );
+		}
+	}
+	this.groups = data.groups;
+};
+
+/**
+ * Fill from template.
+ *
+ * Filling is passive, so existing information is never overwitten. The spec should be re-filled
+ * after a parameter is added to ensure it's still complete, and this is safe because existing data
+ * is never overwritten.
+ */
+ve.dm.WikiaInfoboxSpecModel.prototype.fill = function () {
+	var key;
+
+	for ( key in this.template.getParameters() ) {
+		if ( key && !this.params[key] ) {
+			this.params[key] = this.getDefaultParameterSpec( key );
+		}
+	}
+};
+
+/**
+ * Get the default spec for a parameter.
+ *
+ * @param {string} name Parameter name
+ * @returns {Object} Parameter spec
+ */
+ve.dm.WikiaInfoboxSpecModel.prototype.getDefaultParameterSpec = function ( name ) {
+	return {
+		label: name,
+		default: '',
+		type: 'data',
+		name: name,
+		required: false,
+		deprecated: false
+	};
+};
+
+/**
+ * Get template label.
+ *
+ * @returns {string} Template label
+ */
+ve.dm.WikiaInfoboxSpecModel.prototype.getLabel = function () {
+	var titleObj,
+		title = this.template.getTitle(),
+		target = this.template.getTarget();
+
+	if ( title ) {
+		try {
+			// Normalize and remove namespace prefix if in the Template: namespace
+			titleObj = new mw.Title( title );
+			title = titleObj.getRelativeText( 10 );
```

Comment:
please consider moving `10` namespace to `namespace` variable for better readability